web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1143,31 +1143,65 @@ namespace com.keyman.keyboards {
      *  @param  {string}   cmd        command string
      *  @param  {boolean?} byLanguage if true, context=languages, else context=keyboards
      **/
-    keymanCloudRequest(cmd: string, byLanguage?: boolean) {
-      var URL='https://api.keyman.com/cloud/4.0/', tFlag,
-        Lscript = this.keymanweb.util._CreateElement('script');
-
-      URL = URL + ((arguments.length > 1) && byLanguage ? 'languages' : 'keyboards')
-        +'?jsonp=keyman.register&languageidtype=bcp47&version='+this.keymanweb['version'];
-
+    keymanCloudRequest(cmd: string, byLanguage?: boolean) {    
       var kbdManager = this;
+      var keymanweb = this.keymanweb;
+
+      var TIMEOUT_KEY = "cloud request timed out";
+
+      // Some basic support toward #5044, but definitely not a full solution toward it.
```

Comment:
Can you describe what this is doing here and just leave the `#5044` as a reference for further information? As it stands, I am unclear exactly what we are trying to achieve 😁 

Something like:

```
// Wrap loading of a keyboard script in a Promise so that we can respond
// appropriately on network or parser errors ... etc etc etc
```

And, if the comment is otherwise a TODO, please mark it as such also.